src/virtualenv/util/zipapp.py

```diff
@@ -23,8 +23,10 @@ def extract(full_path, dest):
 
 
 def _get_path_within_zip(full_path):
-    full_path = os.path.abspath(str(full_path))
-    sub_file = full_path[len(ROOT) + 1 :]
+    full_path = os.path.realpath(os.path.abspath(str(full_path)))
+    prefix = ROOT + os.sep
```

Comment:
Can we use here the pathlib variant instead?

Reply:
Sure. I saw most of the other code uses pathlib, but I wasn't sure whether it's intentional or not, and wanted to keep the size of the diff minimal. It's way simpler to implement the whole function using `PurePath.relative_to` anyway.

Question: if I migrate to pathlib, the trivial implementation will throw ValueError in case `ROOT` and `full_path` diverge, i.e. when my assert fails in the current version. Is that acceptable, or should I turn it back to an assert?

Reply:
The ValueError is fine considering we're not expecting it to happen.